src/serial.rs

```diff
@@ -111,6 +116,97 @@ cfg_if! {
     }
 }
 
+/// Types for configuring a serial interface.
+pub mod config {
+    use crate::time::rate::{Baud, Extensions};
+
+    // Reexport stop bit enum from PAC. In case there is a breaking change,
+    // provide a compatible enum from this HAL.
+    pub use crate::pac::usart1::cr2::STOP_A as StopBits;
+
+    /// Parity generation and checking. If odd or even parity is selected, the
+    /// underlying USART will be configured to send/receive the parity bit in
+    /// addtion to the data bits.
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    #[derive(Clone, Copy, PartialEq)]
+    pub enum Parity {
+        /// No parity bit will be added/checked.
+        ParityNone,
+        /// The MSB transmitted/received will be generated/checked to have a
+        /// even number of bits set.
+        ParityEven,
+        /// The MSB transmitted/received will be generated/checked to have a
+        /// odd number of bits set.
+        ParityOdd,
```

Comment:
Why not just

```suggestion
    pub enum Parity {
        /// No parity bit will be added/checked.
        None,
        /// The MSB transmitted/received will be generated/checked to have a
        /// even number of bits set.
        Even,
        /// The MSB transmitted/received will be generated/checked to have a
        /// odd number of bits set.
        Odd,
```

Reply:
I used the scheme above as I have seen it in other STM32 HALs for the sake of compatibility. But I prefer the short versions too.

At a first glance, I thought `None` would be an issue. But this is just a member of an enum and not a keyword.